hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java

```diff
@@ -221,9 +221,6 @@ private long estimateAdditionalCosts(
     private long estimatedGasPriceInTinybars(
             @NonNull final HederaFunctionality functionality, @NonNull final Instant consensusTime) {
         final var feeData = feeManager.getFeeData(functionality, consensusTime, SubType.DEFAULT);
-        if (feeData == null) {
```

Comment:
Would it be better to check against DEFAULT_FEE_DATA, rather than remove this check entirely?

Reply:
sam here

Reply:
As we discussed, this appears to be a case of mismatch between intent-in-code and intent-in-requirements.  The topic will be brought up with the prior author(s) to determine which path is actually correct (default fees will be charged in some cases, or the lack of explicit fee definitions will fail the transaction).  Additional changes are necessary in either case.